build_projects/dotnet-host-build/sign.proj

```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project InitialTargets="SetSigningProperties" DefaultTargets="SignFiles" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <RepoRoot>$(MSBuildThisFileDirectory)/../../</RepoRoot>
+  </PropertyGroup>
+
+  <!-- This will be overriden if we're building with MicroBuild. -->
+  <Target Name="SignFiles">
+    <Message Text="Fake sign target.  Would sign: @(FilesToSign)" />
+  </Target>
+
+  <Import Project="$(RepoRoot)pkg/Tools/MicroBuild.Core.props" />
+  <Import Project="$(RepoRoot)pkg/Tools/MicroBuild.Core.targets" />
+
+  <Target Name="SetSigningProperties">
+    <Error Condition="'$(RID)' == ''" Text="Missing required property 'RID'." />
+    <Error Condition="'$(CertificateId)' == ''" Text="Missing required property 'CertificateId'." />
+    <PropertyGroup>
+      <!-- The OutDir and IntermediateOutputPath properties are required by MicroBuild. MicroBuild only
+           signs files that are under these paths. -->
+      <OutDir Condition="'$(OutDir)' == ''">$(RepoRoot)artifacts/$(RID)/</OutDir>
+      <IntermediateOutputPath Condition="'$(IntermediateOutputPath)' == ''">$(OutDir)</IntermediateOutputPath>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="SignCoreHost" DependsOnTargets="GetSignCoreHostFiles">
+    <CallTarget Targets="SignFiles" />
+  </Target>
+
+  <Target Name="GetSignCoreHostFiles">
+    <ItemGroup>
+      <FilesToSign Include="$(OutDir)corehost/**/hostfxr.dll">
+        <Authenticode>$(CertificateId)</Authenticode>
+      </FilesToSign>
+      <FilesToSign Include="$(OutDir)corehost/**/hostpolicy.dll">
+        <Authenticode>$(CertificateId)</Authenticode>
+      </FilesToSign>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="SignSharedFrameworkCrossgenedAssemblies" DependsOnTargets="GetSignSharedFrameworkCrossgenedAssembliesFiles">
+    <CallTarget Targets="SignFiles" />
+  </Target>
+
+  <Target Name="GetSignSharedFrameworkCrossgenedAssembliesFiles">
+    <ItemGroup>
+      <FilesToSign Include="$(OutDir)intermediate/sharedFrameworkPublish/shared/**/Microsoft.*.dll">
```

Comment:
Should this be what you define IntermediateOutputPath as instead of OutDir?  Not that it matters too much since it's a subfolder.